qualtran/resource_counting/_costing.py

```diff
@@ -0,0 +1,249 @@
+        generalizer: If provided, run this function on each bloq in the call graph to dynamically
+            modify attributes. If the function returns `None`, the bloq is ignored in the
+            cost computation. If a sequence of generalizers is provided, each generalizer
+            will be run in order.
+
+    Returns:
+        The cost value. Its type depends on the provided `cost_key`.
+    """
+    if costs_cache is None:
+        costs_cache = {}
+    if generalizer is None:
+        generalizer = lambda b: b
+    if isinstance(generalizer, (list, tuple)):
+        generalizer = _make_composite_generalizer(*generalizer)
+
+    cost_val = _get_cost_value(bloq, cost_key, costs_cache=costs_cache, generalizer=generalizer)
+    return cost_val
+
+
+def get_cost_cache(
+    bloq: 'Bloq',
+    cost_key: CostKey[CostValT],
+    costs_cache: Optional[Dict['Bloq', CostValT]] = None,
+    generalizer: Optional[Union['GeneralizerT', Sequence['GeneralizerT']]] = None,
+) -> Dict['Bloq', CostValT]:
+    """Build a cache of cost values for the bloq and its callees.
+
+    This can be useful to inspect how callees' costs flow upwards in a given cost computation.
+
+    Args:
+        bloq: The bloq to seed the cost computation.
+        cost_key: A CostKey that specifies which cost to compute.
+        costs_cache: If provided, use this dictionary for initial cached cost values. Values in this
+            dictionary will be preferred over computed values (even if they disagree). This
+            dictionary will be mutated by the function. This dictionary will be returned by the
+            function.
+        generalizer: If provided, run this function on each bloq in the call graph to dynamically
+            modify attributes. If the function returns `None`, the bloq is ignored in the
+            cost computation. If a sequence of generalizers is provided, each generalizer
+            will be run in order.
+
+    Returns:
+        A dictionary mapping bloqs to cost values. The value type depends on the `cost_key`.
+        The bloqs in the mapping depend on the recursive nature of the cost key.
+    """
+    if costs_cache is None:
+        costs_cache = {}
+    if generalizer is None:
+        generalizer = lambda b: b
+    if isinstance(generalizer, (list, tuple)):
+        generalizer = _make_composite_generalizer(*generalizer)
+
+    _get_cost_value(bloq, cost_key, costs_cache=costs_cache, generalizer=generalizer)
+    return costs_cache
+
+
+def query_costs(
+    bloq: 'Bloq',
+    cost_keys: Iterable[CostKey],
+    generalizer: Optional[Union['GeneralizerT', Sequence['GeneralizerT']]] = None,
+) -> Dict['Bloq', Dict[CostKey, CostValT]]:
+    """Compute a selection of costs for a bloq and its callees.
+
+    This function can be used to annotate a call graph diagram with multiple costs
+    for each bloq. Specifically, the return value of this function can be used as the
+    `bloq_data` argument to `GraphvizCallGraph`.
```

Comment:
I see.. you want to force inclusion of bloqs that would not be traversed in the actual computation of the cost. This can always be added as a new function (the body is a simple wrapper over the other functions); but I'd be wary of situations where you'd want to show callee costs that don't actually factor in to their parents' (callers) computation